course_discovery/apps/course_metadata/data_loaders/tests/test_csv_loader.py

```
@@ -601,6 +603,229 @@ def test_ingest_flow_for_preexisting_published_course_with_new_run_creation(self
+                mock_call_api.assert_called_once_with(
+                    "PATCH",
+                    f"http://localhost:18381/api/v1/course/{course_uuid}",
+                    {
+                        "draft": False, "title": "Test Course",
+                        "prices": loader.get_pricing_representation("200", course_type),
+                    },
+                )
+                log_capture.check_present(
+                    (
+                        LOGGER_PATH,
+                        "INFO",
+                        f"Entitlement price update response: {mock_response.content}",
+                    ),
+                )
+                assert response == mock_response.json()
+
+    @responses.activate
+    @mock.patch("course_discovery.apps.course_metadata.data_loaders.csv_loader.download_and_save_course_image")
+    def test_is_logo_downloaded_failure(self, mock_download_image, jwt_decode_patch):  # pylint: disable=unused-argument
+        """
+        Test case to verify that organization_logo_override image download failure is handled properly.
+        """
+        self._setup_prerequisites(self.partner)
+        self.mock_studio_calls(self.partner)
+        self.mock_ecommerce_publication(self.partner)
+        _ = self.mock_image_response()
+        course = CourseFactory(
+            key=self.COURSE_KEY, partner=self.partner, type=self.course_type, draft=True
+        )
+        CourseRunFactory(
+            course=course,
+            key=self.COURSE_RUN_KEY,
+            type=self.course_run_type,
+            status="published",
+            draft=True,
+            variant_id="00000000-0000-0000-0000-000000000000",
+        )
+
+        def download_side_effect(*args, **kwargs):
+            if 'organization_logo_override' in args:
+                return False  # fail for organization_logo_override
+            return True  # Succeed for course_image
```

similar comment as above, we can mock org logo call via responses to return 4xx status. It will make the mocking easier.

---

This is kinda tricky, both the course's `image` and `organization_logo_override` use the same method for `download_image`. In the mock data, the URL for both images is same, and in the `mock_download_image method`, that image_url is hardcoded. That's why I've used it directly with the mock here. If download_image fails for `image`, the continue statement will execute, and it go back to the start.

We can do it too, but it will need some ammendments for this in existing `mock_image_response` method